pkg/webhook/tls/tls.go

```diff
@@ -0,0 +1,95 @@
+package tls
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/go-logr/logr"
+
+	admissionregistrationv1beta1 "k8s.io/api/admissionregistration/v1beta1"
+	certificatesv1beta1 "k8s.io/api/certificates/v1beta1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	certificatesclientv1beta1 "k8s.io/client-go/kubernetes/typed/certificates/v1beta1"
+	"k8s.io/client-go/util/certificate"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+type webhookTLS struct {
+	mgr manager.Manager
+	log logr.Logger
+}
+
+func New(mgr manager.Manager) webhookTLS {
```

Comment:
This is confusing to me. Why is the `type` private and this method public? Due to the private `type`, this cannot be used outside the module anyway, right?

Reply:
this way webhookTLS can only be instanciate with "New" function, so but people can still call it's public methods.